src/sympc/utils/utils.py

```diff
@@ -100,9 +101,15 @@ def wrapper(
         futures = []
         loop = asyncio.get_event_loop()
 
-        with Executor(
-            max_workers=nr_parties, initializer=initializer, initargs=(loop,)
-        ) as executor:
+        ex_kwargs: Dict[str, Any] = {}
+        ex_kwargs["max_workers"] = nr_parties
+
+        # disable additional args in python 3.6
+        if sys.version_info >= (3, 7):
+            ex_kwargs["initializer"] = initializer
```

Comment:
Hmm, and if we have python3.6 I think the Executor would not work? (because it needs that main event loop?)

Reply:
You are right, i think its possible, see comment but I will leave it for the time being and re-address later if we have time.